src/D2L.CodeStyle.Analyzers/ApiUsage/RpcAnalyzer.cs

```diff
@@ -102,6 +129,54 @@ INamedTypeSymbol dependencyAttributeType
 				// other things to check:
 				// - appropriate use of [Dependency]
 			}
+
+			CheckThatRpcParametersAreValid(

```

Comment:
So this should probably move up into the `if` above (and remove the TODO-esque comment).
The same `if` is replicated inside this method in the loop, so that could be removed if it goes into the `if`.

Reply:
Er, not quite... maybe lifting the loop out to here would make more sense

Reply:
We need to check the other parameters regardless of whether `[Dependency]` is available though, yeah?

Reply:
and maybe moving the null check into `IsMarkedDependency`.... I dno, consider it.